storage/csi-cloning.md

```diff
@@ -0,0 +1,105 @@
+
+## Summary
+
+We want to include the upstream CSI Cloning feature in OpenShift. 
+
+## Motivation
+
+* This has been requested by both the OCS and CNV teams
+* This feature is a core functionality offered by storage vendors that we need to enable for our customers
+
+### Goals
+
+* Backport the cloning features in the upstream external-provisioner.
+* Package and ship a downstream image of a sidecar including these features.
+
+### Non-Goals
+
+* This enhancement does NOT propose the addition of other types of DataSource including Populators
+* This enhancement does NOT propose support for special cases like "out of band" cloning (support for back ends that don't have Cloning features), that sort of implementation would fall under Populators.
+* This enhancement does NOT propose any ability to shrink a PVC during a Clone request (e.g. it's considered an invalid request to clone PVC-a with a size of 10Gib to a PVC with a requested size of less than 10Gib, expansion is "ok" if the driver supports it but it's not required)
+* This enhancement does NOT propose adding any ability to transfer a Clone to a different Namespace, the new PVC (Clone) will be in the same Namespace as the origin that was specified. This also means that since this is namespaced, a user can not request a Clone of a PVC that is another Namespace. A user can only request a Clone of PVCs in his or her Namespace.
+* Cloning will only be available in CSI, cloning features will NOT be added to existing in-tree plugins or Flex drivers
+* Cloning will only be available within the same storage class (see [Implementation Details](#implementation-detailsnotesconstraints-optional) section for more info)
+## Proposal
+
+This feature has already been implemented upstream, and is described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20181111-extend-datasource-field.md . It is currently documented at https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/ .
+
+We will be releasing the `csi-external-provisioner` sidecar image with cloning support (already enabled by default) as Technology Preview. This sidecar will be provided to internal teams, such as CNV and OCS, for their drivers. No other consumer is supported, although it is eligible for use by any CSI driver.
+
+### User Stories [optional]
+
+#### Story 1
+As an OCS developer, I want to release the CSI Driver with cloning support so that drivers which include this feature, such as OCS, can use the API to easily Clone a volume.
+
+### Implementation Details/Notes/Constraints [optional]
+
+This feature has already been implemented upstream. Therefore, this feature requires the Kubernetes 1.16 rebase, and the `csi-external-provisioner` to be released with cloning support.
+We must also include e2e tests for this feature.
+
+### Risks and Mitigations
+
+This enhancement requires both of the following:
+
+* Kubernetes 1.16 rebase
+* Upstream release of `csi-external-attacher`. 
```

Comment:
this (and the line below) should use `csi-external-provisioner`

Reply:
I've updated these lines to use `csi-external-provisioner`.